presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -173,6 +173,36 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateErrorTask(
   return std::make_unique<TaskInfo>(info);
 }
 
+std::unique_ptr<TaskInfo> TaskManager::executeResultRequests(const PrestoTaskPtr prestoTask,
+                                        std::unordered_map<int64_t, std::shared_ptr<ResultRequest>>& resultRequests) {
+    try {
+        for (const auto &entry: resultRequests) {
+            auto resultRequest = entry.second.get();
+
+            VLOG(1) << "Processing pending result request for task "
+                    << resultRequest->taskId << ", buffer " << resultRequest->bufferId
+                    << ", sequence " << resultRequest->token;
+            getData(
+                    resultRequest->promise.lock(),
+                    resultRequest->taskId,
+                    resultRequest->bufferId,
+                    resultRequest->token,
+                    resultRequest->maxSize,
+                    *bufferManager_);
+
+        }
+    } catch (const VeloxException& e) {
```

Comment:
This logic is reasonable, but programming by exception is an anti pattern. A better design would be to modify PartitionedOutputBufferManager::getData API to not throw, but indicate task-not-found condition somehow else.